### PR TITLE
Chapter Image settings(back-end):

### PR DESCRIPTION
**Pull request by negulici-r-barnabas**
Chapter Image (Dummy Chapters) can be configured from the front-end by the user.
There are 3 settings:
- duration (seconds)
- count
- resolution (16:9 aspect ratio)

The extraction duration (interval) now can be set in seconds. The minimal value being 1 second. The default value is "300" (5 minutes).
The count is the image file limit / media file, just in case there is bad metadata. Default value is "100".
The resolution sets the extracted image size. It ranges from 144p to 2160p (16:9 aspect ratio). Default value: "Match Source". 

Screenshot of the Dashboard > General settings page:
Screenshot 2022-07-18 175753

Screenshot of the resolution dropdown:

Screenshot 2022-07-18 175845

I tried to keep the changes to minimal. I just used the server configuration that was already implemented by adding 3 more properties to its model class.

I'm a beginner coder. I welcome any improvement suggestions.

The issues that would resolve is  [#1346](https://github.com/jellyfin/jellyfin/issues/1346) (already closed but not implemented).
The feature request is [here](https://features.jellyfin.org/posts/16/allow-custom-chapter-image-intervall).

This is the back-end implementation. The front-end implementation is [here](https://github.com/jellyfin/jellyfin-web/pull/3770).

### CODE REVIEW

**Comment by agrudsky** [comment]
I'm not an approver but just want to say thank you for implementing this! I have wanted this feature for a while.

File: MediaBrowser.Providers/MediaInfo/FFProbeVideoInfo.cs [hunk 1]
```diff
@@ -662,13 +661,13 @@ private ChapterInfo[] CreateDummyChapters(Video video)
                         runtime));
             }
 
-            if (runtime < _dummyChapterDuration)
+            if (runtime < dummyChapterDuration)
             {
                 return Array.Empty<ChapterInfo>();
             }
 
             // Limit to 100 chapters just in case there's some incorrect metadata here
```
Comment (Bond-009) [inline MediaBrowser.Providers/MediaInfo/FFProbeVideoInfo.cs:669]: update comment

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: MediaBrowser.Model/Configuration/ServerConfiguration.cs [hunk 2]
```diff
@@ -240,5 +240,23 @@ public ServerConfiguration()
...
+        /// Gets or sets the dummy chapter count.
+        /// </summary>
+        /// <value>The dummy chapter count.</value>
+        public int DummyChapterCount { get; set; } = 100;
+
+        /// <summary>
+        /// Gets or sets the chapter image resolution.
+        /// </summary>
+        /// <value>The chapter image resolution.</value>
+        public string ChapterImageResolution { get; set; } = "Match Source";
```
[the remark this answers is not in the document]
Comment (crobibero) [reply MediaBrowser.Model/Configuration/ServerConfiguration.cs:260]: I agree, this should be an enum

**Comment by HerrEiklar** [comment]
Status on this?

**Comment by thornbill** [comment]
@HerrEiklar you have feedback here that needs addressed before anything can move forward.

**Comment by negulici-r-barnabas** [comment]
Hello, sorry for my absence I have modified my code with the requested changes.

File: MediaBrowser.Model/Configuration/ServerConfiguration.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (negulici-r-barnabas) [reply MediaBrowser.Model/Configuration/ServerConfiguration.cs]: Hello, could you check my last commit? I changed it to enum.

File: MediaBrowser.Providers/MediaInfo/FFProbeVideoInfo.cs (hunk 1 above)
Comment (negulici-r-barnabas) [reply MediaBrowser.Providers/MediaInfo/FFProbeVideoInfo.cs]: Hello, could you check my last commit? I modified the comment.

File: MediaBrowser.Model/Drawing/ImageResolution.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+{
+    /// <summary>
+    /// Enum ImageResolution.
+    /// </summary>
+    public enum ImageResolution
+    {
+        /// <summary>
+        /// MatchSource.
+        /// </summary>
+        MatchSource,
```
Comment (crobibero) [inline MediaBrowser.Model/Drawing/ImageResolution.cs:11]: Enum values should have associated values
```suggestion
        MatchSource = 0,
```

File: MediaBrowser.Model/Drawing/ImageResolution.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
+namespace MediaBrowser.Model.Drawing
```
Comment (crobibero) [inline MediaBrowser.Model/Drawing/ImageResolution.cs:1]: New files should use [file scoped namespaces](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces)

File: MediaBrowser.Model/Drawing/ImageResolution.cs (hunk 3 above)
Comment (negulici-r-barnabas) [reply MediaBrowser.Model/Drawing/ImageResolution.cs]: I've added the values.

**Review by crobibero — APPROVED** [review]

**Comment by craigers521** [comment]
Awesome! I have been wanting this feature. Can't wait to see this merged!

**Review by Bond-009 — APPROVED** [review]

**Comment by SenorSmartyPants** [comment]
Is there a way to disable dummy chapter creation with this PR? I'd like chapter images for videos that have chapters defined, but I don't want jellyfin to create any dummy chapters at all.

**Comment by craigers521** [comment]
turns out my use case is solved by jellyscrub.  anyone wanting mouse over video preview should check out jellyscrub plugin. https://github.com/nicknsy/jellyscrub

**Comment by Bond-009** [comment]
@SenorSmartyPants setting `DummyChapterCount` to 0 should do the trick

**Comment by Shadowghost** [comment]
#9410 disables it by default and adds a config option to enable it explicitly.

**Comment by Deagan** [comment]
I can't seem to find these settings? Where can I edit these settings on v10.8.9?

**Pull request merged**